api/peering.go

```diff
@@ -96,14 +99,21 @@ func (c *Client) Peerings() *Peerings {
 	return &Peerings{c: c}
 }
 
-func (p *Peerings) Read(ctx context.Context, name string, q *QueryOptions) (*Peering, *QueryMeta, error) {
-	if name == "" {
+func (p *Peerings) Read(ctx context.Context, peeringReq PeeringRequest, q *QueryOptions) (*Peering, *QueryMeta, error) {
+	if peeringReq.Name == "" {
 		return nil, nil, fmt.Errorf("peering name cannot be empty")
 	}
 
-	req := p.c.newRequest("GET", fmt.Sprintf("/v1/peering/%s", name))
+	var url string
+	if peeringReq.Partition != "" {
+		url = fmt.Sprintf("/v1/peering/%s?partition=%s", peeringReq.Name, peeringReq.Partition)
+	} else {
+		url = fmt.Sprintf("/v1/peering/%s", peeringReq.Name)
+	}
```

Comment:
I think the convention is to pass partition through `QueryOptions`. We shouldn't need to use a PeeringRequest